app/src/forms/submission/controller.js

```diff
@@ -62,16 +62,28 @@ module.exports = {
   },
   addStatus: async (req, res, next) => {
     try {
-      const response = await service.changeStatusState(req.params.formSubmissionId, req.body, req.currentUser);
-      // send an email (async in the background)
-      if (req.body.code === Statuses.ASSIGNED && req.body.assignmentNotificationEmail) {
-        const submission = await service.read(req.params.formSubmissionId);
-        emailService.statusAssigned(submission.form.id, response[0], req.body.assignmentNotificationEmail, req.headers.referer).catch(() => { });
-      } else if (req.body.code === Statuses.REVISING && req.body.revisionNotificationEmail) {
-        const submission = await service.read(req.params.formSubmissionId);
-        emailService.statusRevising(submission.form.id, response[0], req.body.revisionNotificationEmail, req.body.revisionNotificationEmailContent, req.headers.referer).catch(() => { });
-      }
-      res.status(200).json(response);
+      const tasks = [
+        service.changeStatusState(req.params.formSubmissionId, req.body, req.currentUser),
+        service.read(req.params.formSubmissionId)
+      ];
+
+      await Promise.all(tasks)
+        .then(results => {
+          const response = results[0];
+          const submission = results[1];
+          // send an email (async in the background)
+          if (req.body.code === Statuses.ASSIGNED && req.body.assignmentNotificationEmail) {
+            emailService.statusAssigned(submission.form.id, response[0], req.body.assignmentNotificationEmail, req.headers.referer).catch(() => { });
+          } else if (req.body.code === Statuses.REVISING && req.body.submissionUserEmail) {
+            emailService.statusRevising(submission.form.id, response[0], req.body.submissionUserEmail, req.body.revisionNotificationEmailContent, req.headers.referer).catch(() => { });
+          } else if (req.body.code === Statuses.COMPLETED && req.body.submissionUserEmail && req.body.confirmCompleted) {
+            emailService.statusCompleted(submission.form.id, response[0], req.body.submissionUserEmail, req.headers.referer).catch(() => { });
+          }
+          res.status(200).json(response);
+        })
+        .catch(error => {
+          next(error);
+        });
```

Comment:
You can avoid writing in promise-chain style since you're already awaiting on the Promise.all - don't need to catch twice. It's generally good practice to either stick with async/await style OR promise-chain style. Mixing both types can cause a lot of grief/confusion down the line with unintended/uncaught behavior.
```js
const results = await Promise.all(tasks);
const response = results[0];
const submission = results[1];
// send an email (async in the background)
if (req.body.code === Statuses.ASSIGNED && req.body.assignmentNotificationEmail) {
  emailService.statusAssigned(submission.form.id, response[0], req.body.assignmentNotificationEmail, req.headers.referer).catch(() => { });
} else if (req.body.code === Statuses.REVISING && req.body.submissionUserEmail) {
  emailService.statusRevising(submission.form.id, response[0], req.body.submissionUserEmail, req.body.revisionNotificationEmailContent, req.headers.referer).catch(() => { });
} else if (req.body.code === Statuses.COMPLETED && req.body.submissionUserEmail && req.body.confirmCompleted) {
  emailService.statusCompleted(submission.form.id, response[0], req.body.submissionUserEmail, req.headers.referer).catch(() => { });
  }

res.status(200).json(response);
```
The trailing `.catch(() => { })` on the emailService calls aren't great practice either; we should consider doing at minimum a `log.warn` of the error, seeing as an email would have failed to dispatch in those cases.